runlabel: execute /proc/self/exe and avoid recursion

vrothberg opened the pull request:
Execute /proc/self/exe instead of podman.  This makes the runlabel
command more portable as it works for binaries outside the path as
well as for local builds.

Also, avoid redundantly executing the runlabel command by setting
the PODMAN_RUNLABEL_NESTED environment variable to "1".  Podman
now checks for this variable before executing the runlabel command
and will throw an error in case the variable is set.

Signed-off-by: Valentin Rothberg <vrothberg@suse.com>

vrothberg commented:
@baude PTAL

vrothberg commented:
Travis is going wild at the moment...

giuseppe commented:
great idea.  LGTM

mheon commented:
/approve

mheon commented on cmd/podman/runlabel.go:
```diff
@@ -94,6 +94,14 @@ func runlabelCmd(c *cli.Context) error {
+	// variable when executing a label first.
+	noexec := os.Getenv("PODMAN_RUNLABEL_NOEXEC")
+	if noexec == "1" {
+		return fmt.Errorf("runlabels cannot execute the runlabel command")
```
Maybe reword this to "cannot execute runlabel because noexec environment variable has been set?

vrothberg replied:
I don't like this one as it doesn't guide the user to resolve the issue. I am also not very happy about the proposed error message but it couldn't find a "better" one that tells the user that recursion isn't allowed.

TomSweeneyRedHat commented on cmd/podman/shared/funcs.go:
```diff
@@ -17,7 +17,7 @@ func GenerateCommand(command, imageName, name string) []string {
 	cmd := strings.Split(command, " ")
 	// Replace the first position of cmd with podman whether
```
I think this comment could use a little tweaking with this change.

baude commented:
LGTM with other people's comments

vrothberg commented:
@baude @mheon @TomSweeneyRedHat thanks for the comments. I changed the error message, updated the comment and changed the tests to reflect the /proc/self/exe change.

baude commented:
/lgtm
